.circleci/config.yml

```diff
@@ -1,92 +1,87 @@
               cp -pr /tmp/workspace/node_modules ~/repo/api/node_modules
               chmod 755 ~/repo/api/node_modules
               chmod 755 ~/repo/api/dist
               docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
-              docker build --build-arg VERSION="${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"${CIRCLE_PROJECT_REPONAME}":"${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"${CIRCLE_PROJECT_REPONAME}":"latest" .
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"${CIRCLE_PROJECT_REPONAME}":"${VERSION}"
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"${CIRCLE_PROJECT_REPONAME}":"latest"
-          name: "Build and Deploy Docker Image"
-      -
-        run:
+              docker build  -t "${DOCKER_REGISTRY}/${NAMESPACE}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_SHA1}" -t "${DOCKER_REGISTRY}/${NAMESPACE}/${CIRCLE_PROJECT_REPONAME}:latest" .
+              docker push "${DOCKER_REGISTRY}/${NAMESPACE}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_SHA1}"
+              docker push "${DOCKER_REGISTRY}/${NAMESPACE}/${CIRCLE_PROJECT_REPONAME}:latest"
+          name: "Build and Deploy API Docker Image"
+      - run: 
           command: ~/repo/scripts/notify_failure.sh
           name: "Report failing Master"
           when: on_fail
     working_directory: ~/repo/api
-  deploy_endpoint:
+
+  deploy_endpoint: 
+    docker: 
+      - image: docker
+    environment: 
+    steps: 
+      - checkout: 
+          path: ~/repo
+      - setup_remote_docker
+      - run: 
+          command: |
+              docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
+              docker build -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_endpoint:${CIRCLE_SHA1}" -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_endpoint:latest" .
+              docker push "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_endpoint:${CIRCLE_SHA1}"
+              docker push "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_endpoint:latest"
+          name: "Build and Deploy Endpoint Docker Image"
+    working_directory: ~/repo/extract_endpoint
+
+  deploy_etl:
     docker:
-      -
-        image: docker
+      - image: docker
     environment:
-      -
-        VERSION: "0.1.3"
     steps:
-      -
-        checkout:
+      - checkout:
           path: ~/repo
       - setup_remote_docker
-      -
-        run:
+      - run:
           command: |
               docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
-              docker build --build-arg VERSION="${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"latest" .
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"${VERSION}"
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"latest"
+              docker build -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_etl:${CIRCLE_SHA1}" -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_etl:latest" .
```

Comment:
These are all so similar, it would seem they could all be a single bash file with different arguments passed in.